client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -21,121 +20,184 @@ class PostRelativeTime extends React.PureComponent {
+				} ),
+				sameElse: this.props.translate( '[for] ll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - "at" and "for" is translated',
+				} ),
+			} );
+		} else {
+			if ( Math.abs( now.diff( this.getTimestamp(), 'days' ) ) < 7 ) {
+				const time = moment( this.getTimestamp() );
+				return time.fromNow();
+			}
+
+			let sameElseTranslation;
+			if ( ! prefix ) {
+				sameElseTranslation = this.props.translate( 'll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
+				} );
+			} else {
+				sameElseTranslation = this.props.translate( '[on] ll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" and "on" is translated',
+				} );
+			}
 
-		const { status, modified, date } = post;
-		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
+			displayedTime = timestamp.calendar( null, {
+				sameElse: sameElseTranslation,
+			} );
 		}
 
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		// If the content is scheduled to be release within a year, do not display the year at the end
+		return timestamp.diff( now, 'years' ) > 0
+			? displayedTime
+			: displayedTime.replace( timestamp.format( 'Y' ), '' );
 	}
 
 	getTimeText() {
 		const time = this.getTimestamp();
-		if ( ! time ) {
-			return null;
-		}
-
 		return (
 			<span className="post-relative-time-status__time">
-				<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
-				<time className="post-relative-time-status__time-text" dateTime={ time }>
-					{ this.getDisplayedTimeFromPost( this.props.post ) }
-				</time>
+				{ time && (
+					<>
+						<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
+						<time className="post-relative-time-status__time-text" dateTime={ time }>
+							{ this.getDisplayedTimeForLabel( true ) }
+						</time>
+					</>
+				) }
 			</span>
 		);
 	}
 
+	/**
+	 * Get status label
+	 *
+	 */
 	getStatusText() {
 		const status = this.props.post.status;
-		let statusClassName = 'post-relative-time-status__status';
+		let extraStatusClassName;
 		let statusIcon = 'aside';
 		let statusText;
 
-		if ( this.props.post.sticky ) {
-			statusText = this.props.translate( 'sticky' );
-			statusClassName += ' is-sticky';
-			statusIcon = 'bookmark-outline';
-		} else if ( status === 'pending' ) {
-			statusText = this.props.translate( 'pending review' );
-			statusClassName += ' is-pending';
+		if ( status === 'trash' ) {
+			extraStatusClassName = 'is-trash';
+			statusIcon = 'trash';
+			const displayedTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'trashed %(displayedTime)s', {
```

Comment:
Cannot add more comments to the [previous discussion](https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415657120), so I'm copying the comments over here to continue it.

@mmtr https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415657120:

> @Automattic/i18n or @akirk can confirm, but I think we need to handle both status and date on the same translation. For instance, here `displayScheduleTime` contains `on [date]` for posts trashed more than a week ago, so `trashed %(displayScheduleTime)s` finally reads as `trashed on [date]`. Some languages might need to move the translated "on" part before "trashed" (while leaving the date after). To support those languages, the translatable string should be `trashed on [date]` rather than `trashed [datePrefixedWithOn]`.
>
> The same applies for the rest of statuses (scheduled for [date], published on [date], draft last modified on [date], ...).

@millerf https://github.com/Automattic/wp-calypso/pull/41301#discussion_r415989089:

> This is quite complex to tackle, as the "on" is not always present (for example : "Published yesterday" vs "Published on Mar, 23 march 2020")
And as translation texts cannot have variables to pass on the status, I would have to have to list all cases that could happen.
> 
> Also waiting to translators to confirm, but I think the "on" is the beginning of the proposition of time, so it correlates more with the date than the adjective. As in "On 23rd of March, I published, etc,..."  But I might be mistaken...

Reply:
> Also waiting to translators to confirm, but I think the "on" is the beginning of the proposition of time, so it correlates more with the date than the adjective

Yeah, let's see if @Automattic/i18n or @akirk can jump in. I might be wrong as well, but languages are so different that I wouldn't assume that prepositions are always part of the date in all of them.